ql/pricingengines/swap/crossccyswapengine.cpp

```diff
@@ -0,0 +1,165 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2016 Quaternion Risk Management Ltd
+ All rights reserved.
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+#include <ql/cashflows/cashflows.hpp>
+#include <ql/exchangerate.hpp>
+#include <ql/utilities/dataformatters.hpp>
+
+#include <ql/pricingengines/swap/crossccyswapengine.hpp>
+
+namespace QuantLib {
+
+CrossCcySwapEngine::CrossCcySwapEngine(const Currency& ccy1, const Handle<YieldTermStructure>& currency1Discountcurve,
+                                       const Currency& ccy2, const Handle<YieldTermStructure>& currency2Discountcurve,
```

Comment:
Any better name for instead of currency1/currency2?  Domestic/foreign?  Base/other?  Main/other?  @pcaspers, was there any reason you picked 1 and 2?